### ISSUE TITLE
`read-write` test failed and cannot be killed

### ISSUE THREAD

**Issue by ammarfaizi2** [axboe/liburing#313]
Good day,
I have an issue with `read-write` test. The test is failing and cannot be killed, this causes the whole system cannot be turned off or suspend at all. I had to unplug the machine power or long pressing the power button to force shut it down every time I run this test.

So `systemd suspend -i` and `poweroff` don't work after I run this test.

Below is the detailed information:
## Kernel Info
```
ammarfaizi2@integral:~$ uname -a
Linux integral 5.8.0-25-generic #26-Ubuntu SMP Thu Oct 15 10:30:38 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux
ammarfaizi2@integral:~$ 
```
## GCC version
```
ammarfaizi2@integral:~$ cc --version
cc (Ubuntu 10.2.1-6ubuntu1) 10.2.1 20210110
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```

## Setup
```
cd /tmp;
git clone https://github.com/axboe/liburing;
cd liburing;
git checkout liburing-2.0;
make -j4;
```

## Run the test (failed and hang, won't exit)
```
ammarfaizi2@integral:/tmp/liburing$ test/read-write
SQPOLL requires root, skipping
Not root, skipping test_write_efbig
wait_cqe=-4
test_rem_buf async failed
^C^C^C^\^\^\ # Hang here...
```

## Attempt to kill, but not killed
```
ammarfaizi2@integral:~$ ps aux -AT | grep read-write
ammarfa+    3173    3173  0.0  0.0      0     0 pts/1    D+   14:43   0:00 [read-write]
ammarfa+    3607    3607  0.0  0.0   6508  2736 pts/4    S+   14:49   0:00 grep --color=auto read-write
ammarfaizi2@integral:~$ for i in {1..10}; do sudo pkill -SIGKILL -e read-write; done;
read-write killed (pid 3173)
read-write killed (pid 3173)
read-write killed (pid 3173)
read-write killed (pid 3173)
read-write killed (pid 3173)
read-write killed (pid 3173)
read-write killed (pid 3173)
read-write killed (pid 3173)
read-write killed (pid 3173)
read-write killed (pid 3173)
ammarfaizi2@integral:~$ ps aux -AT | grep read-write
ammarfa+    3173    3173  0.0  0.0      0     0 pts/1    D+   14:43   0:00 [read-write]
ammarfa+    3629    3629  0.0  0.0   6508  2628 pts/4    S+   14:50   0:00 grep --color=auto read-write
ammarfaizi2@integral:~$ 
```

I tried to strace to see what happened, but result in `Operation not permitted`.
I checked from `htop`, the process still exists, with zero memory, with state `D`, zero CPU usage.
## Strace
```
ammarfaizi2@integral:~$ sudo strace -ivtp 3173
strace: Could not attach to process. If your uid matches the uid of the target process, check the setting of /proc/sys/kernel/yama/ptrace_scope, or try again as the root user. For more details, see /etc/sysctl.d/10-ptrace.conf: Operation not permitted
strace: attach: ptrace(PTRACE_SEIZE, 3173): Operation not permitted
ammarfaizi2@integral:~$ 
```

## /proc/3173/maps is empty
```
ammarfaizi2@integral:~$ sudo cat /proc/3173/maps
ammarfaizi2@integral:~$ 
```

## /proc/3173/syscall
```
ammarfaizi2@integral:~$ sudo cat /proc/3173/syscall
231 0x1 0xe7 0x3c 0x4 0xffffffffffffff88 0x1 0x7ffe00c6f7e8 0x7f381ad38d21
ammarfaizi2@integral:~$ 
```

## dmesg
```
[   87.096416] Bridge firewalling registered
[   88.799090] Initializing XFRM netlink socket
[   90.672901] Bluetooth: RFCOMM TTY layer initialized
[   90.672918] Bluetooth: RFCOMM socket layer initialized
[   90.672931] Bluetooth: RFCOMM ver 1.11
[  726.312910] INFO: task io_wqe_worker-0:3433 blocked for more than 120 seconds.
[  726.312919]       Not tainted 5.8.0-25-generic #26-Ubuntu
[  726.312922] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[  726.312926] io_wqe_worker-0 D    0  3433      2 0x00004000
[  726.312933] Call Trace:
[  726.312951]  __schedule+0x212/0x5d0
[  726.312958]  schedule+0x55/0xc0
[  726.312963]  schedule_preempt_disabled+0xe/0x10
[  726.312967]  __mutex_lock.constprop.0+0x14a/0x490
[  726.312975]  ? io_issue_sqe+0xdd9/0x1150
[  726.312979]  __mutex_lock_slowpath+0x13/0x20
[  726.312982]  mutex_lock+0x34/0x40
[  726.312986]  io_issue_sqe+0x108b/0x1150
[  726.312991]  io_wq_submit_work+0x68/0x1c2
[  726.312996]  ? _cond_resched+0x1a/0x50
[  726.313001]  io_worker_handle_work+0x199/0x300
[  726.313006]  io_wqe_worker+0x16a/0x1b0
[  726.313012]  kthread+0x12f/0x150
[  726.313016]  ? io_worker_exit+0x190/0x190
[  726.313020]  ? __kthread_bind_mask+0x70/0x70
[  726.313026]  ret_from_fork+0x22/0x30
[  726.313032] INFO: task io_wqe_worker-0:3434 blocked for more than 120 seconds.
[  726.313036]       Not tainted 5.8.0-25-generic #26-Ubuntu
[  726.313039] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[  726.313042] io_wqe_worker-0 D    0  3434      2 0x00004000
[  726.313046] Call Trace:
[  726.313051]  __schedule+0x212/0x5d0
[  726.313056]  schedule+0x55/0xc0
[  726.313061]  schedule_preempt_disabled+0xe/0x10
[  726.313064]  __mutex_lock.constprop.0+0x14a/0x490
[  726.313072]  ? dbs_update_util_handler+0x1b/0x80
[  726.313078]  ? attach_entity_load_avg+0x1be/0x1f0
[  726.313082]  __mutex_lock_slowpath+0x13/0x20
[  726.313085]  mutex_lock+0x34/0x40
[  726.313089]  io_issue_sqe+0x1020/0x1150
[  726.313095]  io_wq_submit_work+0x68/0x1c2
[  726.313100]  ? _cond_resched+0x1a/0x50
[  726.313105]  io_worker_handle_work+0x199/0x300
[  726.313110]  io_wqe_worker+0x16a/0x1b0
[  726.313114]  kthread+0x12f/0x150
[  726.313118]  ? io_worker_exit+0x190/0x190
[  726.313122]  ? __kthread_bind_mask+0x70/0x70
[  726.313127]  ret_from_fork+0x22/0x30
[  726.313133] INFO: task io_wqe_worker-1:3435 blocked for more than 120 seconds.
[  726.313136]       Not tainted 5.8.0-25-generic #26-Ubuntu
[  726.313139] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[  726.313142] io_wqe_worker-1 D    0  3435      2 0x00004000
[  726.313145] Call Trace:
[  726.313151]  __schedule+0x212/0x5d0
[  726.313156]  schedule+0x55/0xc0
[  726.313160]  schedule_preempt_disabled+0xe/0x10
[  726.313164]  __mutex_lock.constprop.0+0x14a/0x490
[  726.313169]  ? default_send_IPI_single+0x37/0x40
[  726.313173]  ? resched_curr+0x5e/0xc0
[  726.313177]  __mutex_lock_slowpath+0x13/0x20
[  726.313180]  mutex_lock+0x34/0x40
[  726.313184]  io_issue_sqe+0x1020/0x1150
[  726.313190]  io_wq_submit_work+0x68/0x1c2
[  726.313194]  ? _cond_resched+0x1a/0x50
[  726.313209]  io_worker_handle_work+0x199/0x300
[  726.313215]  io_wqe_worker+0x16a/0x1b0
[  726.313220]  kthread+0x12f/0x150
[  726.313225]  ? io_worker_exit+0x190/0x190
[  726.313230]  ? __kthread_bind_mask+0x70/0x70
[  726.313235]  ret_from_fork+0x22/0x30
[  726.313242] INFO: task io_wqe_worker-0:3436 blocked for more than 120 seconds.
[  726.313246]       Not tainted 5.8.0-25-generic #26-Ubuntu
[  726.313249] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[  726.313252] io_wqe_worker-0 D    0  3436      2 0x00004000
[  726.313257] Call Trace:
[  726.313263]  __schedule+0x212/0x5d0
[  726.313269]  schedule+0x55/0xc0
[  726.313274]  schedule_preempt_disabled+0xe/0x10
[  726.313278]  __mutex_lock.constprop.0+0x14a/0x490
[  726.313284]  __mutex_lock_slowpath+0x13/0x20
[  726.313288]  mutex_lock+0x34/0x40
[  726.313293]  io_issue_sqe+0x1020/0x1150
[  726.313299]  io_wq_submit_work+0x68/0x1c2
[  726.313304]  ? _cond_resched+0x1a/0x50
[  726.313310]  io_worker_handle_work+0x199/0x300
[  726.313317]  io_wqe_worker+0x16a/0x1b0
[  726.313322]  kthread+0x12f/0x150
[  726.313327]  ? io_worker_exit+0x190/0x190
[  726.313331]  ? __kthread_bind_mask+0x70/0x70
[  726.313336]  ret_from_fork+0x22/0x30
[  726.313343] INFO: task io_wqe_worker-1:3438 blocked for more than 120 seconds.
[  726.313348]       Not tainted 5.8.0-25-generic #26-Ubuntu
[  726.313351] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[  726.313354] io_wqe_worker-1 D    0  3438      2 0x00004000
[  726.313358] Call Trace:
[  726.313365]  __schedule+0x212/0x5d0
[  726.313371]  schedule+0x55/0xc0
[  726.313376]  schedule_preempt_disabled+0xe/0x10
[  726.313380]  __mutex_lock.constprop.0+0x14a/0x490
[  726.313385]  ? default_send_IPI_single+0x37/0x40
[  726.313389]  ? resched_curr+0x5e/0xc0
[  726.313393]  __mutex_lock_slowpath+0x13/0x20
[  726.313397]  mutex_lock+0x34/0x40
[  726.313402]  io_issue_sqe+0x1020/0x1150
[  726.313409]  io_wq_submit_work+0x68/0x1c2
[  726.313414]  ? _cond_resched+0x1a/0x50
[  726.313420]  io_worker_handle_work+0x199/0x300
[  726.313426]  io_wqe_worker+0x16a/0x1b0
[  726.313431]  kthread+0x12f/0x150
[  726.313436]  ? io_worker_exit+0x190/0x190
[  726.313441]  ? __kthread_bind_mask+0x70/0x70
[  726.313446]  ret_from_fork+0x22/0x30
[  726.313453] INFO: task io_wqe_worker-0:3439 blocked for more than 120 seconds.
[  726.313457]       Not tainted 5.8.0-25-generic #26-Ubuntu
[  726.313460] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[  726.313463] io_wqe_worker-0 D    0  3439      2 0x00004000
[  726.313467] Call Trace:
[  726.313473]  __schedule+0x212/0x5d0
[  726.313479]  schedule+0x55/0xc0
[  726.313484]  schedule_preempt_disabled+0xe/0x10
[  726.313488]  __mutex_lock.constprop.0+0x14a/0x490
[  726.313494]  __mutex_lock_slowpath+0x13/0x20
[  726.313498]  mutex_lock+0x34/0x40
[  726.313503]  io_issue_sqe+0x1020/0x1150
[  726.313510]  io_wq_submit_work+0x68/0x1c2
[  726.313515]  ? _cond_resched+0x1a/0x50
[  726.313521]  io_worker_handle_work+0x199/0x300
[  726.313527]  io_wqe_worker+0x16a/0x1b0
[  726.313532]  kthread+0x12f/0x150
[  726.313537]  ? io_worker_exit+0x190/0x190
[  726.313542]  ? __kthread_bind_mask+0x70/0x70
[  726.313547]  ret_from_fork+0x22/0x30
[  726.313554] INFO: task io_wqe_worker-1:3440 blocked for more than 120 seconds.
[  726.313558]       Not tainted 5.8.0-25-generic #26-Ubuntu
[  726.313561] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[  726.313564] io_wqe_worker-1 D    0  3440      2 0x00004000
[  726.313568] Call Trace:
[  726.313575]  __schedule+0x212/0x5d0
[  726.313581]  schedule+0x55/0xc0
[  726.313586]  schedule_preempt_disabled+0xe/0x10
[  726.313590]  __mutex_lock.constprop.0+0x14a/0x490
[  726.313595]  ? default_send_IPI_single+0x37/0x40
[  726.313599]  ? resched_curr+0x5e/0xc0
[  726.313604]  __mutex_lock_slowpath+0x13/0x20
[  726.313607]  mutex_lock+0x34/0x40
[  726.313613]  io_issue_sqe+0x1020/0x1150
[  726.313619]  io_wq_submit_work+0x68/0x1c2
[  726.313624]  ? _cond_resched+0x1a/0x50
[  726.313630]  io_worker_handle_work+0x199/0x300
[  726.313637]  io_wqe_worker+0x16a/0x1b0
[  726.313642]  kthread+0x12f/0x150
[  726.313647]  ? io_worker_exit+0x190/0x190
[  726.313651]  ? __kthread_bind_mask+0x70/0x70
[  726.313657]  ret_from_fork+0x22/0x30
[  726.313663] INFO: task io_wqe_worker-0:3441 blocked for more than 120 seconds.
[  726.313667]       Not tainted 5.8.0-25-generic #26-Ubuntu
[  726.313670] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[  726.313673] io_wqe_worker-0 D    0  3441      2 0x00004000
[  726.313677] Call Trace:
[  726.313683]  __schedule+0x212/0x5d0
[  726.313690]  schedule+0x55/0xc0
[  726.313695]  schedule_preempt_disabled+0xe/0x10
[  726.313699]  __mutex_lock.constprop.0+0x14a/0x490
[  726.313705]  __mutex_lock_slowpath+0x13/0x20
[  726.313708]  mutex_lock+0x34/0x40
[  726.313713]  io_issue_sqe+0x1020/0x1150
[  726.313720]  io_wq_submit_work+0x68/0x1c2
[  726.313725]  ? _cond_resched+0x1a/0x50
[  726.313731]  io_worker_handle_work+0x199/0x300
[  726.313737]  io_wqe_worker+0x16a/0x1b0
[  726.313742]  kthread+0x12f/0x150
[  726.313747]  ? io_worker_exit+0x190/0x190
[  726.313752]  ? __kthread_bind_mask+0x70/0x70
[  726.313757]  ret_from_fork+0x22/0x30
[  726.313764] INFO: task io_wqe_worker-1:3442 blocked for more than 120 seconds.
[  726.313768]       Not tainted 5.8.0-25-generic #26-Ubuntu
[  726.313771] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[  726.313774] io_wqe_worker-1 D    0  3442      2 0x00004000
[  726.313778] Call Trace:
[  726.313784]  __schedule+0x212/0x5d0
[  726.313790]  schedule+0x55/0xc0
[  726.313796]  schedule_preempt_disabled+0xe/0x10
[  726.313800]  __mutex_lock.constprop.0+0x14a/0x490
[  726.313805]  ? default_send_IPI_single+0x37/0x40
[  726.313809]  ? resched_curr+0x5e/0xc0
[  726.313813]  __mutex_lock_slowpath+0x13/0x20
[  726.313817]  mutex_lock+0x34/0x40
[  726.313822]  io_issue_sqe+0x1020/0x1150
[  726.313829]  io_wq_submit_work+0x68/0x1c2
[  726.313834]  ? _cond_resched+0x1a/0x50
[  726.313840]  io_worker_handle_work+0x199/0x300
[  726.313846]  io_wqe_worker+0x16a/0x1b0
[  726.313851]  kthread+0x12f/0x150
[  726.313856]  ? io_worker_exit+0x190/0x190
[  726.313860]  ? __kthread_bind_mask+0x70/0x70
[  726.313866]  ret_from_fork+0x22/0x30
[  726.313873] INFO: task io_wqe_worker-0:3443 blocked for more than 120 seconds.
[  726.313877]       Not tainted 5.8.0-25-generic #26-Ubuntu
[  726.313880] "echo 0 > /proc/sys/kernel/hung_task_timeout_secs" disables this message.
[  726.313883] io_wqe_worker-0 D    0  3443      2 0x00004000
[  726.313886] Call Trace:
[  726.313893]  __schedule+0x212/0x5d0
[  726.313899]  schedule+0x55/0xc0
[  726.313904]  schedule_preempt_disabled+0xe/0x10
[  726.313908]  __mutex_lock.constprop.0+0x14a/0x490
[  726.313915]  ? dbs_update_util_handler+0x1b/0x80
[  726.313920]  ? attach_entity_load_avg+0x1be/0x1f0
[  726.313925]  __mutex_lock_slowpath+0x13/0x20
[  726.313928]  mutex_lock+0x34/0x40
[  726.313934]  io_issue_sqe+0x1020/0x1150
[  726.313940]  io_wq_submit_work+0x68/0x1c2
[  726.313946]  ? _cond_resched+0x1a/0x50
[  726.313951]  io_worker_handle_work+0x199/0x300
[  726.313958]  io_wqe_worker+0x16a/0x1b0
[  726.313963]  kthread+0x12f/0x150
[  726.313968]  ? io_worker_exit+0x190/0x190
[  726.313972]  ? __kthread_bind_mask+0x70/0x70
[  726.313978]  ret_from_fork+0x22/0x30
[ 1224.381062] PM: suspend entry (deep)
[ 1229.787993] Filesystems sync: 5.416 seconds
[ 1229.969968] Freezing user space processes ... 
[ 1249.954948] Freezing of tasks failed after 20.007 seconds (1 tasks refusing to freeze, wq_busy=0):
[ 1249.955253] read-write      D    0  3173   1954 0x80004007
[ 1249.955261] Call Trace:
[ 1249.955282]  __schedule+0x212/0x5d0
[ 1249.955290]  schedule+0x55/0xc0
[ 1249.955297]  schedule_preempt_disabled+0xe/0x10
[ 1249.955301]  __mutex_lock.constprop.0+0x14a/0x490
[ 1249.955307]  __mutex_lock_slowpath+0x13/0x20
[ 1249.955311]  mutex_lock+0x34/0x40
[ 1249.955319]  io_ring_ctx_wait_and_kill+0x3e/0x2e0
[ 1249.955325]  io_uring_release+0x20/0x30
[ 1249.955331]  __fput+0xe9/0x250
[ 1249.955336]  ____fput+0xe/0x10
[ 1249.955342]  task_work_run+0x6d/0xa0
[ 1249.955347]  do_exit+0x200/0x440
[ 1249.955352]  do_group_exit+0x3b/0xb0
[ 1249.955356]  __x64_sys_exit_group+0x18/0x20
[ 1249.955362]  do_syscall_64+0x49/0xc0
[ 1249.955368]  entry_SYSCALL_64_after_hwframe+0x44/0xa9
[ 1249.955374] RIP: 0033:0x7f381ad38d21
[ 1249.955388] Code: Bad RIP value.
[ 1249.955391] RSP: 002b:00007ffe00c6f7e8 EFLAGS: 00000246 ORIG_RAX: 00000000000000e7
[ 1249.955396] RAX: ffffffffffffffda RBX: 00007f381ae3d610 RCX: 00007f381ad38d21
[ 1249.955399] RDX: 000000000000003c RSI: 00000000000000e7 RDI: 0000000000000001
[ 1249.955402] RBP: 0000000000000001 R08: ffffffffffffff88 R09: 0000000000000001
[ 1249.955404] R10: 0000000000000004 R11: 0000000000000246 R12: 00007f381ae3d610
[ 1249.955406] R13: 0000000000000001 R14: 00007f381ae3dae8 R15: 0000000000000000
[ 1249.955427] OOM killer enabled.
[ 1249.955428] Restarting tasks ... done.
[ 1250.164139] PM: suspend exit
[ 1250.164336] PM: suspend entry (s2idle)
[ 1250.656549] Filesystems sync: 0.492 seconds
[ 1250.673862] Freezing user space processes ... 
[ 1270.669004] Freezing of tasks failed after 20.008 seconds (1 tasks refusing to freeze, wq_busy=0):
[ 1270.669310] read-write      D    0  3173   1954 0x80004007
[ 1270.669317] Call Trace:
[ 1270.669338]  __schedule+0x212/0x5d0
[ 1270.669346]  schedule+0x55/0xc0
[ 1270.669353]  schedule_preempt_disabled+0xe/0x10
[ 1270.669358]  __mutex_lock.constprop.0+0x14a/0x490
[ 1270.669364]  __mutex_lock_slowpath+0x13/0x20
[ 1270.669367]  mutex_lock+0x34/0x40
[ 1270.669375]  io_ring_ctx_wait_and_kill+0x3e/0x2e0
[ 1270.669381]  io_uring_release+0x20/0x30
[ 1270.669387]  __fput+0xe9/0x250
[ 1270.669392]  ____fput+0xe/0x10
[ 1270.669398]  task_work_run+0x6d/0xa0
[ 1270.669404]  do_exit+0x200/0x440
[ 1270.669408]  do_group_exit+0x3b/0xb0
[ 1270.669413]  __x64_sys_exit_group+0x18/0x20
[ 1270.669419]  do_syscall_64+0x49/0xc0
[ 1270.669424]  entry_SYSCALL_64_after_hwframe+0x44/0xa9
[ 1270.669431] RIP: 0033:0x7f381ad38d21
[ 1270.669445] Code: Bad RIP value.
[ 1270.669448] RSP: 002b:00007ffe00c6f7e8 EFLAGS: 00000246 ORIG_RAX: 00000000000000e7
[ 1270.669453] RAX: ffffffffffffffda RBX: 00007f381ae3d610 RCX: 00007f381ad38d21
[ 1270.669455] RDX: 000000000000003c RSI: 00000000000000e7 RDI: 0000000000000001
[ 1270.669458] RBP: 0000000000000001 R08: ffffffffffffff88 R09: 0000000000000001
[ 1270.669461] R10: 0000000000000004 R11: 0000000000000246 R12: 00007f381ae3d610
[ 1270.669463] R13: 0000000000000001 R14: 00007f381ae3dae8 R15: 0000000000000000
[ 1270.669482] OOM killer enabled.
[ 1270.669483] Restarting tasks ... done.
[ 1270.898411] PM: suspend exit
```

I would be happy to provide more detailed information if required. Just tell me what should I do and send.

**Comment by ammarfaizi2** [comment]
It seems this issue has been fixed on Linux 5.12-rc3.
```
root@h222p:~/liburing# uname -r
5.12.0-051200rc3-generic
root@h222p:~/liburing# test/read-write
root@h222p:~/liburing# echo $?
0
root@h222p:~/liburing# 
```

**Comment by ammarfaizi2** [comment]
I am closing this issue.
